### ISSUE TITLE
Player to Object attachments are buggy at high velocity.

### ISSUE THREAD

**Issue by ElCeejo** [minetest/minetest#10739]
I've noticed at high velocity a player attached to entity lag behind. Using `vector.distance()` I noticed the player can read well over 100 nodes away from the entity it's attached to. This is an issue because if the player is attached to a fast moving object and the player lags too far behind the entity will unload. There is no pretty fix for this aside from making objects slower which isn't optimal.

To recreate:

Create and entity and attach the player to it, then set velocity to a high value. Within 30 minutes are pretty high the player will detach, though it may take longer.

**Comment by randomMesh** [comment]
I noticed this too when traveling in a minecart and then, all of a sudden, i got dismounted behind the trails and fell to my death.

**Comment by SmallJoker** [comment]
The player position seems to lag behind one server step:

**Comment by lhofhansl** [comment]
I noticed that at low velocities as well. Occasionally the view angle would jump for a few frames and then get back to normal. And overall the movement while being attached seems choppy.

You can try that nicely with the hot air balloon mod.

**Comment by lhofhansl** [comment]
Worse, if anti-cheat is on (disabled in single player mode) a distance of just 4 nodes will reset the attached object to the player position. If server step is slow, this will happen most of the time.

**Comment by SmallJoker** [comment]
@lhofhansl Positions are updated in bulk in every step. So it might be possible that the attachment position is updated after the player, hence causing a delay of a server tick. It's just a speculation according to the code, as I'm not sure how to reliably prove that.

**Comment by lhofhansl** [comment]
I tried to order the step() calls to do entities first and players last, but it did not fix the problem I am seeing.
Part of the problem with a laggy server is that we limit the dtime to 0.5s when doing collision detection (see also #10825).
When the player is attached this becomes a problem.

**Comment by TheTermos** [comment]
Since 5.0 my experience has been rougly the same
- player as a 1st level attachment - fine
- player as a 2nd level attachment - detaching
- entity-entity attachments - fine regardless of level

Now I hear even 1st level attachments are broken? I couldn't reproduce but it may be different on laggy servers.
This needs to be investigated for reals.

**Comment by MisterE123** [comment]
I have a definite example of buggy detachment with 1st level attachment.

to replicate, use MT 5.3.0
Use vanilla MTG with the following mods:
arena_lib: https://gitlab.com/zughy-friends-minetest/arena_lib/-/tree/master
signs_lib: https://content.minetest.net/packages/VanessaE/signs_lib/
unified_inventory
world_edit
wormball : https://github.com/MisterE123/wormball-WIP-   --- USE THE DEBUG BRANCH!


Follow the instructions in readme.txt of wormball to create an arena, and then play that arena in singleplayer.

You will notice that quite often, the player is detached.
the debug branch also includes chat messages that indicate the player's attached state, as well as the line of code that they were attached/detached with.

Theory behind the wormball minigame: (a snake clone)

at the start of the minigame, a dummy entity (wormball:att) is created at the location of a worm head and the player is attached to this entity. THen, every .2 sec, (using an on_step) the worm head is advanced, and the att ent is gotten from the player, and then the ent's move_to function is called, to move the ent to the new head location. 

FOr the debug version, I have added the following to the ent's definition:
`
    on_attach_child = function(self, child)
        if child:is_player() then
            minetest.chat_send_all(child:get_player_name()..' was attached')
        end

    end,
    on_detach_child = function(self, child)
        if child:is_player() then
            minetest.chat_send_all(child:get_player_name()..' was detached')
        end

    end,


`
Using the output of these debug messages:
-----------------------------
WHAT SHOULD HAPPEN:
1) player joins game
2) ent "wormball:att" is created, player is attached to ent. Chat message is sent:
"Player was attached" -- from the on_child_attach of the ent
3)every second, using the minigame library's on_time_tick function, the following message is sent, indicating that the player is still attached to the ent:
"time: <num> player <playername> is attached"
4)the player crashes the snake or wins, and the player is detached 
"player was detached" -- from the on_child_detach of the ent
and then the att ent is removed.
---------------------------
WHAT DOES HAPPEN (often enough):
1) player joins game
2) ent "wormball:att" is created, player is attached to ent. Chat message is sent:
"Player was attached" -- from the on_child_attach of the ent
3)every second, using the minigame library's on_time_tick function, the following message is sent, indicating that the player is still attached to the ent:
"time: <num> player <playername> is attached"
4) the player is suddenly teleported to the place they wore before they were attached
5) they are then quickly placed at the current location of the att ent
6)every second, the minigame library's on_time_tick function now reports about the player
"time: <num> player <playername> is not attached"
HOWEVER, we never hear form the entity's on_child_detach callback!
7) the player now has free movement inside the arena. They are not moved with the snake head. THey find it hard to control the snake, and quickly lose.   ]:(

----
I am uncertain whether the entity remains in existence all the time or not. Certainly, it does remain in existence some of the times the bug occurs. I tested this by putting a chat message in the on_step
